core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -228,12 +234,13 @@ public void seek(long pos) throws IOException {
       return;
     }
     Preconditions.checkArgument(pos >= 0, PreconditionMessage.ERR_SEEK_NEGATIVE, pos);
-    Preconditions.checkArgument(validPosition(pos),
-        PreconditionMessage.ERR_SEEK_PAST_END_OF_FILE, pos);
-
-    seekBlockInStream(pos);
-    checkAndAdvanceBlockInStream();
-    mCurrentBlockInStream.seek(mPos % mBlockSize);
+    Preconditions.checkArgument(
+        pos < mFileLength, PreconditionMessage.ERR_SEEK_PAST_END_OF_FILE, pos);
```

Comment:
Just to explain what I meant: 
validPosition only checks pos < mFileLength, we still need to keep the other check for pos >= 0.  we generate ERR_SEEK_PAST_END_OF_FILE error msg for validPosition check failure which assumes that validPositions checking pos < mFileLength.  So I think if we use validPosition, it is better to change validPostion to include pos >=0 and update the error msgs.

but I don't really have strong opinions on this. I have changed it back.